lib/Backends/CPU/DebugInfo.cpp

```diff
@@ -80,6 +80,9 @@ llvm::DIType *LLVMIRGen::getDebugType(llvm::IRBuilder<> &builder,
 void LLVMIRGen::generateFunctionDebugInfo(llvm::Function *F) {
   if (!emitDebugInfo)
     return;
+  // Bail if debug information was emitted for this function already.
```

Comment:
Why do we need this? Do we run the debug generation code multiple times?

Reply:
Fixed. We should not run the debug info generation multiple times. There is an assert now to check that it never happens.